lib/core/src/persist/mod.rs

```diff
@@ -306,35 +328,30 @@ impl Persister {
             "INSERT INTO payment_tx_data (
            tx_id,
            timestamp,
-           asset_id,
-           amount,
            fees_sat,
-           payment_type,
            is_confirmed,
            unblinding_data
         )
-        VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+        VALUES (?, ?, ?, ?, ?)
         ON CONFLICT (tx_id)
         DO UPDATE SET timestamp = CASE WHEN excluded.is_confirmed = 1 THEN excluded.timestamp ELSE timestamp END,
-                      asset_id = excluded.asset_id,
-                      amount = excluded.amount,
                       fees_sat = excluded.fees_sat,
-                      payment_type = excluded.payment_type,
                       is_confirmed = excluded.is_confirmed,
                       unblinding_data = excluded.unblinding_data
         ",
             (
                 &ptx.tx_id,
                 ptx.timestamp.or(Some(utils::now())),
-                ptx.asset_id,
-                ptx.amount,
                 ptx.fees_sat,
-                ptx.payment_type,
                 ptx.is_confirmed,
                 ptx.unblinding_data,
             ),
         )?;
 
+        for balance in balances {
+            Self::insert_or_update_payment_balance(&tx, &ptx.tx_id, balance)?;
```

Comment:
should we delete the balances before updating in case the asset id no longer exists?

Reply:
I couldn't find where the delete was happening